zero_bin/common/src/prover_state/mod.rs

```diff
@@ -242,6 +243,7 @@ impl ProverStateManager {
     /// Initialize global prover state from the configuration.
     pub fn initialize(&self) -> anyhow::Result<()> {
         info!("initializing prover state...");
+        set_circuit_cache_dir_env_if_not_set()?;
```

Comment:
Both the `leader` & `worker` need to persist circuits, so I thought it made sense to do the env logic inside common. Let me know if anyone prefers this to be in the individual binaries.